addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/handler/DSMRBridgeHandler.java

```diff
@@ -0,0 +1,299 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dsmr.handler;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.dsmr.internal.device.DSMRDevice;
+import org.openhab.binding.dsmr.internal.device.DSMRDeviceConfiguration;
+import org.openhab.binding.dsmr.internal.device.DSMRDeviceRunnable;
+import org.openhab.binding.dsmr.internal.device.DSMREventListener;
+import org.openhab.binding.dsmr.internal.device.DSMRFixedConfigDevice;
+import org.openhab.binding.dsmr.internal.device.DSMRSerialAutoDevice;
+import org.openhab.binding.dsmr.internal.device.connector.DSMRConnectorErrorEvent;
+import org.openhab.binding.dsmr.internal.device.connector.DSMRSerialSettings;
+import org.openhab.binding.dsmr.internal.device.connector.SerialPortManager;
+import org.openhab.binding.dsmr.internal.device.p1telegram.P1Telegram;
+import org.openhab.binding.dsmr.internal.device.p1telegram.P1TelegramListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DSMRBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author M. Volaart - Initial contribution
+ * @author Hilbrand Bouwkamp - Refactored way messages are forwarded to meters. Removed availableMeters dependency.
+ */
+@NonNullByDefault
+public class DSMRBridgeHandler extends BaseBridgeHandler implements DSMREventListener {
+
+    /**
+     * Factor that will be multiplied with {@link #receivedTimeoutNanos} to get the timeout factor after which the
+     * device is set off line.
+     */
+    private static final int OFFLINE_TIMEOUT_FACTOR = 10;
+
+    private final Logger logger = LoggerFactory.getLogger(DSMRBridgeHandler.class);
+
+    /**
+     * Additional meter listeners to get received meter values.
+     */
+    private final List<P1TelegramListener> meterListeners = new ArrayList<>();
+
+    /**
+     * Serial Port Manager.
+     */
+    private final SerialPortManager serialPortManager = new SerialPortManager() {
+    };
+
+    /**
+     * The dsmrDevice managing the connection and handling telegrams.
+     */
+    @Nullable
+    private DSMRDevice dsmrDevice;
+
+    /**
+     * Long running process that controls the DSMR device connection.
+     */
+    @Nullable
+    private DSMRDeviceRunnable dsmrDeviceRunnable;
```

Comment:
Annotate the type with `@Nullable` instead:

```java
private @Nullable DSMRDeviceRunnable dsmrDeviceRunnable;
```

Please also fix all the other occurrences of this issue (also in other classes). :-)